server/src/main/java/io/druid/server/lookup/cache/LookupCoordinatorManager.java

```diff
@@ -506,137 +274,426 @@ public boolean deleteLookup(final String tier, final String lookup, AuditInfo au
+
+    if (allLookupTiers == null) {
+
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("Not updating lookups because no data exists");
+      }
+      return;
+    }
+
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("Starting lookup sync for on all nodes.");
+    }
+
+    final Map<HostAndPort, LookupsStateWithMap> currState = new HashMap<>();
+    final Random rand = new Random();
+
+    try {
+      List<ListenableFuture<?>> futures = new ArrayList<>();
+      for (String tier : allLookupTiers.keySet()) {
+
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("Starting lookup mgmt for tier [%s].", tier);
+        }
+
+        final Map<String, LookupExtractorFactoryMapContainer> tierLookups = allLookupTiers.get(tier);
+        for (final HostAndPort node : listenerDiscoverer.getNodes(LookupModule.getTierListenerPath(tier))) {
+
+          if (LOG.isDebugEnabled()) {
+            LOG.debug("Starting lookup mgmt for tier [%s] and host [%s:%s].", tier, node.getHostText(), node.getPort());
           }
+
+          futures.add(
+              executorService.submit(
+                  new Runnable()
+                  {
+                    @Override
+                    public void run()
+                    {
+                      try {
+
+                        if (LOG.isDebugEnabled()) {
+                          LOG.debug("Starting lookup sync for node [%s].", node);
+                        }
+
+                        LookupsStateWithMap lookupsState = knownOldState.get() != null ? knownOldState.get().get(node) : null;
+                        if (lookupsState == null
+                            || !lookupsState.getToLoad().isEmpty()
+                            || !lookupsState.getToDrop().isEmpty()
+                            || rand.nextBoolean()) {
+                          lookupsState = getLookupStateForNode(node);
+
+                          if (LOG.isDebugEnabled()) {
+                            LOG.debug("Received lookups state from node [%s].", node);
+                          }
+                        }
+                        currState.put(node, lookupsState);
+
+                        Map<String, LookupExtractorFactoryMapContainer> toLoad = new HashMap<>();
+                        for (Map.Entry<String, LookupExtractorFactoryMapContainer> e : tierLookups.entrySet()) {
+                          String name = e.getKey();
+                          LookupExtractorFactoryMapContainer lookupToBe = e.getValue();
+
+                          LookupExtractorFactoryMapContainer current = lookupsState.getToLoad().get(name);
+                          if (current == null) {
+                            current = lookupsState.getCurrent().get(name);
+                          }
+
+                          if (current == null || lookupToBe.replaces(current)) {
+                            toLoad.put(name, lookupToBe);
+                          }
```

Comment:
`current` tells what is currently at the node,  if `lookupToBe` doesn't replace `current` then no update is required .... it will happen if node already has the `lookupToBe` loaded (eg. coordinator did it in last loop, and user hasn't changed that lookup version since)